From b7654530cf55a226c6aa6a9991b2b130c26325fa Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Wed, 3 Aug 2016 10:55:41 -0500
Subject: [PATCH 1147/5242] MLK-13182-3: ARM: dts: imx enable egalax_ts
 support for imx6q pop arm2

commit  f8629519de2437c5cad7c78245b0ce42b2195f0f from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable egalax_ts touch screen support for imx6q pop arm2
target platform.

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6q-pop-arm2.dts |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx6q-pop-arm2.dts b/arch/arm/boot/dts/imx6q-pop-arm2.dts
index 65e5958..3d0a23f 100644
--- a/arch/arm/boot/dts/imx6q-pop-arm2.dts
+++ b/arch/arm/boot/dts/imx6q-pop-arm2.dts
@@ -174,6 +174,16 @@
 		compatible = "fsl,imx6-hdmi-i2c";
 		reg = <0x50>;
 	};
+
+	egalax_ts@04 {
+		compatible = "eeti,egalax_ts";
+		reg = <0x04>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_egalax_int>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <31 2>;
+		wakeup-gpios = <&gpio3 31 0>;
+	};
 };
 
 &i2c3 {
@@ -255,6 +265,12 @@
 			>;
 		};
 
+		pinctrl_egalax_int: egalax_intgrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D31__GPIO3_IO31 	0x80000000
+			>;
+		};
+
 		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
 				MX6QDL_PAD_GPIO_9__PWM1_OUT 	0x1b0b1
-- 
1.7.9.5

